### PR TITLE
Add effortToFix, Support newline character in comment, Add comment for column/endColumn

### PR DESCRIPTION
**Pull request by alban-auzeill**

### CODE REVIEW

File: test-commons/README.md [hunk 1]
```diff
@@ -81,12 +81,15 @@ fileIssues.addComment(line, column, commentText);
...
 fileIssues.addActualIssue(lineNumber, message, null);
-// or with precise location:
+// or with precise location (note: column start at 1 and endColumn is the last character of the token,
+// not the following one, if the token has only one character, endColumn == column)
 UnderlinedRange primaryRange = new UnderlinedRange(from.line(), from.column(), to.endLine(), to.endColumn());
 PrimaryLocation primary = new PrimaryLocation(primaryRange, secondaryLocationCount);
 primary.addSecondary(secondaryRange, secondarymessage);
 fileIssues.addActualIssue(lineNumber, message, primary);
 
+// Note: Issue global to a file can be report at line 0 and asserted using "// Noncompliant@0 {{...}}"
```
Comment (vilchik-elena) [inline test-commons/README.md]: @alban-auzeill typo: `reportED`

File: test-commons/src/main/java/com/sonarsource/checks/verifier/LineIssues.java [hunk 2]
```diff
@@ -141,7 +141,8 @@ public String toString() {
     }
     Double effort = effortToFix();
     if (effort != null) {
-      out.append(" [[effortToFix=").append(effort.longValue()).append("]]");
+      String displayedEffort = effort.toString().replaceFirst(".0$", "");
```
Comment (vilchik-elena) [inline test-commons/src/main/java/com/sonarsource/checks/verifier/LineIssues.java]: @alban-auzeill why `.replaceFirst(".0$", "")`?

File: test-commons/src/main/java/com/sonarsource/checks/verifier/NoncompliantCommentParser.java [hunk 3]
```diff
@@ -41,7 +41,7 @@
     + "(?<messages>(\\{\\{.*?\\}\\} *+)+)?"
     // params, ex: [[effortToFix=2;id=main]]
     + "(?:\\[\\[(?<params>[^\\]]++)\\]\\] *+)?"
-    + "(\r|\n|\r\n)?");
+    + "(?:\r(\n?)|\n)?");
```
Comment (vilchik-elena) [inline test-commons/src/main/java/com/sonarsource/checks/verifier/NoncompliantCommentParser.java]: @alban-auzeill why this change?

File: test-commons/src/main/java/com/sonarsource/checks/verifier/LineIssues.java (hunk 2 above)
Comment (alban-auzeill) [reply test-commons/src/main/java/com/sonarsource/checks/verifier/LineIssues.java]: Error . should be replaced by \\.
